### DISCUSSION TITLE
TreeViewItem set ItemsSource ,but it's childs not binding attr successed

### DISCUSSION THREAD

**Discussion by liufei134** [AvaloniaUI/Avalonia#12470] [Q&A]
I want use TreeView control, inside of it's children ,treeviewItem ,I set it's attr 'ItemsSource' and set it's ItemTemplate ,but i cann't get my goal , following code below:
`    <TreeView >
      <TreeViewItem Header="1112" ItemsSource="{Binding Persons}">
        <TreeViewItem.ItemTemplate>
          <DataTemplate >
            <TextBlock Text="{Binding Name}"></TextBlock>
          </DataTemplate>
        </TreeViewItem.ItemTemplate>
      </TreeViewItem>
    <TreeView.ItemsPanel>
      <ItemsPanelTemplate>
        <VirtualizingStackPanel/>
      </ItemsPanelTemplate>
    </TreeView.ItemsPanel>
  </TreeView>`

it cann't work well

**Comment by timunie** [comment]
Hey @liufei134 

this isn't a bug imo but more a question you have. Please ask these on telegram or in github discussions. If you consider this to be a bug, please fill out the templates provided. Thx. 

I'll convert it into a discussion now for you.

**Comment by timunie** [comment]
@liufei134 it's better to use the MVVM approach here and define the ItemSource inside your ViewModel. Then you can setup a TreeDataTemplate to manage your tree nodes. Inside this, one can bind to child collections. See: https://docs.avaloniaui.net/docs/next/reference/controls/detailed-reference/treeview-1
